src/dialogs/copyPasteOwnership.tsx

```diff
@@ -0,0 +1,221 @@
+
+        // "original" is a special value, valid values are formatted as 'user:group'
+        if (ownerStr !== "original") {
+            await cockpit.spawn([
+                "chown",
+                "--recursive",
+                ownerStr,
+                ...clipboard.files.map(file => dstPath + "/" + file.name),
+            ], { superuser: "require" });
+        }
+    } catch (err) {
+        const e = err as cockpit.BasicError;
+        addAlert(_("Pasting failed"), AlertVariant.danger, `${new Date().getTime()}`, e.message);
+
+        // cleanup potentially copied files in case of "chown" fail
+        try {
+            await cockpit.spawn([
+                "rm",
+                "-rf",
+                ...clipboard.files.map(file => dstPath + "/" + file.name)
+            ], { superuser: "try" });
+        } catch (ex) {
+            console.warn(`Failed to clean up copied files in ${dstPath}`, ex);
+        }
+    }
+}
+
+function ownershipTypes(files: FolderFileInfo[]) {
+    const owners = new Set();
+
+    for (const file of files) {
+        const user = file.user;
+        const group = file.group;
+
+        if (user === group) {
+            owners.add(String(user));
+        } else {
+            owners.add(`${user}:${group}`);
+        }
+    }
+
+    return [...owners];
+}
+
+function makeCandidatesMap(currentUser: cockpit.UserInfo, cwdInfo: FileInfo, clipboard: ClipboardInfo) {
+    // keys in Map() are ordered in the same order as they were inserted
+    // and has no duplicates. Only the first insertion.
+    const map: Map<string, string> = new Map();
+    const candidates = get_owner_candidates(currentUser, cwdInfo);
+
+    // also add current ownership if it is same for all files (shallow check)
+    const firstFile = clipboard.files[0];
+    const firstOwnerStr = `${firstFile.user}:${firstFile.group}`;
+    const uniqueOwners = ownershipTypes(clipboard.files);
+    if (uniqueOwners.length === 1) {
+        candidates.add(firstOwnerStr);
+    }
+
+    candidates.forEach(owner => {
+        const split = owner.split(':');
+        let key = owner;
+        if (split.length === 2 && split[0] === split[1]) {
+            key = split[0];
+        }
+        if (uniqueOwners.length === 1 && owner === firstOwnerStr) {
+            key = `${key} ${_("(original owner)")}`;
+        }
+
+        map.set(key, owner);
+    });
+
+    // add option to keep the same permissions on mixed permissions files
+    if (uniqueOwners.length > 1) {
+        const dots = uniqueOwners.length > 2 ? ", ..." : "";
+        map.set(`${_("keep original owners")} (${uniqueOwners.slice(0, 2).join(", ")}${dots})`, "original");
+    }
+
+    return map;
+}
+
+const CopyPasteAsOwnerModal = ({
+    clipboard,
+    dialogResult,
+    path,
+} : {
+    clipboard: ClipboardInfo,
+    dialogResult: DialogResult<void>,
+    path: string,
+}) => {
+    // @ts-expect-error superuser.js is not typed
+    useEvent(superuser, "changed");
```

Comment:
I just realized that I misread that the previous time -- this modal is a short-lived thing, it's unlikely that the superuser status changes underneath it. However, modals don't actually block the superuser widget.

However, this is still a no-op -- with that, the modal gets re-rendered on superuser changes, but useInit() doesn't re-run. I'd say let's not make this overly complicated and just drop this useEvent().